infrastructures/openshift/src/main/java/org/eclipse/che/workspace/infrastructure/openshift/multiuser/oauth/OpenshiftTokenInitializationFilter.java

```diff
@@ -78,44 +79,54 @@ public OpenshiftTokenInitializationFilter(
   }
 
   @Override
-  protected String getUserId(String token) {
-    try {
-      io.fabric8.openshift.api.model.User user = getCurrentUser(token);
-      return firstNonNull(user.getMetadata().getUid(), user.getMetadata().getName());
-    } catch (KubernetesClientException e) {
-      if (e.getCode() == 401) {
-        LOG.error(
-            "Unauthorized when getting current user. Invalid OpenShift token, probably expired. Re-login? Re-request the token?");
-      }
-      throw new RuntimeException(e);
-    } catch (InfrastructureException e) {
-      throw new RuntimeException(e);
-    }
+  protected io.fabric8.openshift.api.model.User processToken(String token) {
+    return getCurrentUser(token);
+  }
+
+  @Override
+  protected String getUserId(String token, io.fabric8.openshift.api.model.User user) {
+    return firstNonNull(user.getMetadata().getUid(), user.getMetadata().getName());
   }
 
   @Override
-  protected Subject extractSubject(String token) {
+  protected @Nullable Subject extractSubject(
+      String token, io.fabric8.openshift.api.model.User osu) {
     try {
-      ObjectMeta userMeta = getCurrentUser(token).getMetadata();
+      ObjectMeta userMeta = osu.getMetadata();
       User user =
           userManager.getOrCreateUser(
-              firstNonNull(userMeta.getUid(), userMeta.getName()),
-              openshiftUserEmail(userMeta),
-              userMeta.getName());
+              getUserId(token, osu), openshiftUserEmail(userMeta), userMeta.getName());
       return new AuthorizedSubject(
           new SubjectImpl(user.getName(), user.getId(), token, false), permissionChecker);
-    } catch (InfrastructureException | ServerException | ConflictException e) {
+    } catch (ServerException | ConflictException e) {
       throw new RuntimeException(e);
     }
   }
 
-  private io.fabric8.openshift.api.model.User getCurrentUser(String token)
-      throws InfrastructureException {
+  /**
+   * Returns the current user or null if the token cannot be authorized.
+   *
+   * @param token the token to authorize to OpenShift with
+   * @return the current OpenShift user or null
+   * @throws KubernetesClientException when the request to get the current user fails
+   */
+  private @Nullable io.fabric8.openshift.api.model.User getCurrentUser(String token)
```

Comment:
do we need this method? looks like it's called only from `processToken`, so we can move the code there directly